dali/test/dali_operator_test.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_TEST_DALI_OPERATOR_TEST_H_
+#define DALI_TEST_DALI_OPERATOR_TEST_H_
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <memory>
+#include <dali/pipeline/pipeline.h>
+#include <dali/test/op_graph.h>
+#include <dali/test/tensor_list_wrapper.h>
+
+namespace dali {
+namespace testing {
+
+using Arguments = std::map<std::string, double>; // TODO: some generalization. boost::any?
```

Comment:
This is still a stub and does not satisfy our API requirements:
1. Obviously, we need some kind of generalization.
2. We need two-part key for this mapping. We can use a single string for that, but then must document how this string is parsed to address both graph node name and argument name.

Suggested solutions:
1.
```
struct ArgumentKey : std::pair<std::string, std::string> {
  using base = std::pair<std::string, std::string>;
  ArgumentKey(std::string arg_name) : base({}, std::move(arg_name)) {}  // NOLINT
  ArgumentKey(std::string node_name, std::string arg_name) : base(std::move(node_name),  std::move(arg_name)) {}
  const std::string &node_name() const { return first; }
  const std::string &arg_name() const { return second; }
};
```
2. Just use string (`"node:arg"`). Not specifying node (`"arg"`) will assume ther's just one operator. Pipeline build should fail if it's not the case.

Reply:
Done